cmd/fluxsvc/main.go

```diff
@@ -181,11 +184,25 @@ func main() {
 	}
 
 	// Release workers.
+	waitForWorkers := make(chan struct{})
+	shutdownWorkers := make(chan struct{})
 	{
-		worker := release.NewWorker(rjs, instancer, releaseMetrics, logger)
-		releaseTicker := time.NewTicker(time.Second)
-		defer releaseTicker.Stop()
-		go worker.Work(releaseTicker.C)
+		workers := sync.WaitGroup{}
+
+		// for each worker goroutine
+		{
+			workers.Add(1)
+			worker := release.NewWorker(rjs, instancer, releaseMetrics, logger)
+			go func() {
+				defer workers.Done()
+				worker.Work(shutdownWorkers)
+			}()
```

Comment:
What about encapsulating the goroutines inside the worker? so it just has a single poller looking at the db, then passes jobs off to goroutines?

A Worker.Stop() method could then do the shutdown/wait dance.

WDYT?